capstone-frontend/src/App.js

```diff
@@ -1,25 +1,41 @@
 import React from 'react';
-import logo from './logo.svg';
+import {
+  Route, 
+  BrowserRouter as Router
+} from 'react-router-dom';
+
 import './App.css';
 
+import Home from './components/Home';
+import Browse from './components/Browse';
+import MyBooks from './components/MyBooks';
+import MyLists from './components/MyLists';
+import MyClubs from './components/MyClubs';
+import BookPage from './components/BookPage';
+import ListPage from './components/ListPage';
+
+import Navbar from './components/Navbar';
+import LeftSideBar from './components/LeftSideBar';
+import RightSideBar from './components/RightSideBar';
+
 function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Router>
+      <Navbar />
+      <div className="row">
+        <LeftSideBar />
+        <div className="col-12 col-md-8" id="body-row">
+          <Route exact path='/' component={Home} />
+          <Route path='/browse' component={Browse} />
+          <Route path='/mybooks' component={MyBooks} />
+          <Route path='/mylists' component={MyLists} />
+          <Route path='/myclubs' component={MyClubs} />
+          <Route path='/bookdetails' render={(props) => <BookPage {...props} />} />
+          <Route path='/listdetails' render={(props) => <ListPage {...props} />} />
+        </div>
```

Comment:
What does this line of code do specifically? ```render={(props) => <BookPage {...props} />}```

Reply:
This line will allow us to send props to the BookPage. Say you click on a SearchBookItem (as rendered in search). It will have a link that sends the gbookID to the BookPage, this is possible as a result of sending these props.